ravenpy/models/emulators/hypr.py

```diff
@@ -0,0 +1,611 @@
+from collections import defaultdict
+from pathlib import Path
+from typing import cast
+
+import xarray as xr
+from pydantic.dataclasses import dataclass
+
+from ravenpy.config.commands import (
+    HRU,
+    LU,
+    BaseDataCommand,
+    BasinIndexCommand,
+    HRUState,
+    Sub,
+)
+from ravenpy.config.rvs import RVH, RVI
+from ravenpy.models.base import Ostrich, Raven
+
+
+class HYPR(Raven):
+    """
+    The HYdrological model for Prairie Region (HYPR) is based on the
+    conceptual Hydrologiska Byrans Vattenbalansavdelning (HBV)-light
+    model. HBV is modified to work in the prairies by incorporating a
+    conceptual lateral-flow component to represent the pothole storage
+    complexities. HYPR can be used for prairie streamflow simulation.
+
+    References
+    ----------
+
+    Mohamed I. Ahmed, Amin Elshorbagy, and Alain Pietroniro (2020).
+    Toward Simple Modeling Practices in the Complex Canadian Prairie
+    Watersheds. Journal of Hydrologic Engineering, 25(6), 04020024.
+    doi: 10.1061/(ASCE)HE.1943-5584.0001922.
+    """
+
+    @dataclass
+    class Params:
+        par_x01: float
+        par_x02: float
+        par_x03: float
+        par_x04: float
+        par_x05: float
+        par_x06: float
+        par_x07: float
+        par_x08: float
+        par_x09: float
+        par_x10: float
+        par_x11: float
+        par_x12: float
+        par_x13: float
+        par_x14: float
+        par_x15: float
+        par_x16: float
+        par_x17: float
+        par_x18: float
+        par_x19: float
+        par_x20: float
+        par_x21: float
+
+    @dataclass
+    class DerivedParams:
+        POW_X05: float
+        POW_X06: float
+
+    @dataclass
+    class HRU(HRU):  # type: ignore
+        land_use_class: str = "OPEN_1"
+        veg_class: str = "FOREST"
+        soil_profile: str = "DEFAULT_P"
+        aquifer_profile: str = "[NONE]"
+        terrain_class: str = "[NONE]"
+
+    def __init__(self, *args, **kwds):
+        super().__init__(*args, **kwds)
+
+        self.config.update(
+            identifier="hypr",
+            hrus=(HYPR.HRU(),),
+            subbasins=(
+                Sub(
+                    subbasin_id=1,
+                    name="sub_001",
+                    downstream_id=-1,
+                    profile="None",
+                    gauged=True,
+                ),
+            ),
+        )
+
+        #########
+        # R V P #
+        #########
+
+        rvp_tmpl = """
+        # tied parameters:
+        # (it is important for OSTRICH to find every parameter place holder somewhere in this file)
+        # (without this "para_x05" and "para_x06" wouldn't be detectable)
+        #    para_pow_x5     = {derived_params.POW_X05}     =  10^(para_x05)       = 10^{params.par_x05}
+        #    para_pow_x6     = {derived_params.POW_X06}     =  10^(para_x06)       = 10^{params.par_x06}
```

Comment:
This does not work. Both values (`derived_params.POW_X05/06` and `params.par_x05/06`) get filled in with same value...

Reply:
I don't understand what you mean, those are two seperate variables, and it's a comment. You mean the comment is not generated properly?

Reply:
That's correct; it is a comment. But it is setup like you can see the sampled and the derived parameters such that one can easier check what actually happened and it is really convenient for debugging and checking. HERE: the derived parameter is POW_X05 (which is 10**x05).

So, it would be great to see (let's say x05 is sampled to be 2):
`para_pow_x5     = 100.0    =  10^(para_x05)       = 10^2.0`

You see what I mean? It worked in the examples I provided. This might happen at more places. This is one where I spotted it

Reply:
I went back in your own code. In your test you have this:

https://github.com/CSHS-CWRA/RavenPy/blob/add-hypr-model/tests/test_hypr.py#L30-L31

so it seems (as you say in the comments) that `par_x05` and `par_x06` are actually 10 to their power, from the start.. which would make sense because later you have this:

https://github.com/CSHS-CWRA/RavenPy/blob/add-hypr-model/ravenpy/models/emulators.py#L718-L720

(I always wondered what was the point of having the part on the right commented out, because without it `par_x05` is just trivially equal to `POW_X05`). I asked @huard but he could not explain it to me. I have observed that you have the same kind of logic in a few other places.

So the result is that `par_x05` == `POW_X05` (idem for `06`), from start to finish, and so when RavenPy renders the template values that you have in your RVP, namely:

https://github.com/CSHS-CWRA/RavenPy/blob/add-hypr-model/ravenpy/models/raven-hypr/raven-hypr.rvp#L13-L14

the resulting RVP is, when I run it (which I just did, using the code in your PR):

```
#    para_pow_x5     = 0.4650976     =  10^(para_x05)       = 10^0.4650976
#    para_pow_x6     = 0.1177004     =  10^(para_x06)       = 10^0.1177004
```

which does not make sense to me. I realize it's probably me who is confused, but please help me understand what I'm getting wrong.

Reply:
Just to clarify: these are RV comments so they do not affect the actual results in any way of course, but still the comments do not make sense (arithmetically speaking).

Reply:
`par_x05` is NOT equal to `POW_X05`. That is VERY important. When we just run Raven, we pass a set of parameters- original one AND derived ones. When we use OSTRICH, all parameters are the original ones and Ostrich is doing the derivation of the parameters. Hence all the comments in my codes to make clear if something is derived or not. The piece above is for the forward run of Raven (not Ostrich) and hence you are right, `par_x05` is not actually known. So leave it as it is. But please don't rename POW_x05 to par_x05 because that is not correct.

Reply:
This all won't change the results since this is all comments. But you can see how confusing this already is to you, so it will be even more confusing to the users if there is written:
`0.4650976 = 10^(para_x05) = 10^0.4650976`. We need to remove the last piece then and make it only `0.4650976 = 10^(para_x05)`. Makes sense?